### PR TITLE
ALL: uniformize verify_command

### PR DESCRIPTION
**Pull request by khancyr**
Coming from unreachable return false in APMrover2/command_logic.cpp L181, and questioning why verify_command return true if command is not verify, I unify the 3 versions since they are doing the same thing

### CODE REVIEW

**Comment by khancyr** [comment]
Previously copter didn't have warning message on commands handling, was it intentional ?

**Comment by rmackay9** [comment]
Looks reasonable.  Maybe we should send the command number down to the GCS when it fails.  I think there's a gcs_send_text_fmt() or similar.  The message is a bit long perhaps.  I'd probably shorten it to "Skipping invalid cmd %d"

**Comment by gmorph** [comment]
Reviewing now.

File: ArduPlane/commands_logic.cpp [hunk 1]
```diff
@@ -251,12 +252,12 @@ bool Plane::verify_command(const AP_Mission::Mission_Command& cmd)        // Ret
     case MAV_CMD_NAV_TAKEOFF:
         return verify_takeoff();
 
-    case MAV_CMD_NAV_LAND:
-        return verify_land();
-
     case MAV_CMD_NAV_WAYPOINT:
         return verify_nav_wp(cmd);
 
+    case MAV_CMD_NAV_LAND:
```
Comment (gmorph) [inline ArduPlane/commands_logic.cpp]: Why was this moved to below MAV_CMD_NAV_WAYPOINT?

File: APMrover2/commands_logic.cpp [hunk 2]
```diff
@@ -170,15 +175,32 @@ bool Rover::verify_command(const AP_Mission::Mission_Command& cmd)
...
+        case MAV_CMD_DO_CONTROL_VIDEO:
+        case MAV_CMD_DO_DIGICAM_CONFIGURE:
+        case MAV_CMD_DO_DIGICAM_CONTROL:
+        case MAV_CMD_DO_SET_CAM_TRIGG_DIST:
+        case MAV_CMD_DO_SET_ROI:
+        case MAV_CMD_DO_SET_REVERSE:
+            return true;
+
         default:
-            if (cmd.id > MAV_CMD_CONDITION_LAST) {
```
Comment (magicrub) [inline APMrover2/commands_logic.cpp]: this should probably stay

File: ArduPlane/commands_logic.cpp [hunk 3]
```diff
@@ -306,16 +306,16 @@ bool Plane::verify_command(const AP_Mission::Mission_Command& cmd)        // Ret
...
-    case MAV_CMD_DO_DIGICAM_CONFIGURE:
-    case MAV_CMD_DO_DIGICAM_CONTROL:
-    case MAV_CMD_DO_SET_CAM_TRIGG_DIST:
-    case MAV_CMD_NAV_ROI:
-    case MAV_CMD_DO_MOUNT_CONFIGURE:
     case MAV_CMD_DO_INVERTED_FLIGHT:
     case MAV_CMD_DO_LAND_START:
     case MAV_CMD_DO_FENCE_ENABLE:
     case MAV_CMD_DO_AUTOTUNE_ENABLE:
+    case MAV_CMD_DO_CONTROL_VIDEO:
```
Comment (gmorph) [inline ArduPlane/commands_logic.cpp]: Was there a reason these were moved?

File: ArduPlane/commands_logic.cpp [hunk 4]
```diff
@@ -278,6 +279,12 @@ bool Plane::verify_command(const AP_Mission::Mission_Command& cmd)        // Ret
     case MAV_CMD_NAV_ALTITUDE_WAIT:
         return verify_altitude_wait(cmd);
 
+    case MAV_CMD_NAV_VTOL_TAKEOFF:
```
Comment (gmorph) [inline ArduPlane/commands_logic.cpp]: Why were these moved?

File: APMrover2/commands_logic.cpp (hunk 2 above)
Comment (gmorph) [inline APMrover2/commands_logic.cpp]: yes agreed

**Comment by gmorph** [comment]
Its a good PR - the Rover catchall has been on my list so really happy its been done.  If you could just clarify a couple of points I reckon we can get it in.  Thanks!

File: APMrover2/commands_logic.cpp (hunk 2 above)
Comment (OXINARF) [inline APMrover2/commands_logic.cpp]: He's using *is_nav_cmd*, which is what Plane uses. He tried to make all vehicles consistent (which seems like a good thing to me).

**Comment by khancyr** [comment]
@gmorph I move some case for putting then in the same order that in start_command function.

File: APMrover2/commands_logic.cpp (hunk 2 above)
Comment (khancyr) [inline APMrover2/commands_logic.cpp]: In fact, I change MAV_CMD_CONDITION_LAST because it accept all command without notify. Or not all commands are defined/used.
We should improve the if with : 
<MAV_CMD_NAV_LAST
<MAV_CMD_CONDITION_LAST
<MAV_CMD_DO_LAST
in that way we catch all possibility and notify if it is possible to be use or not. 
What do you think ?

**Comment by khancyr** [comment]
Agreed with @rmackay9 suggestion to use 
gcs_send_text_fmt(MAV_SEVERITY_WARNING,"Skipping invalid cmd #%i",cmd.id)

**Comment by magicrub** [comment]
This is flagged as needsRework, is that still accurate?

**Comment by khancyr** [comment]
@magicrub yes. I need to update with randy suggestion. 
I was waiting for answer to : 
"In fact, I change MAV_CMD_CONDITION_LAST because it accept all command without notify. Or, not all commands are defined/used.
We should improve the if with : 
<MAV_CMD_NAV_LAST
<MAV_CMD_CONDITION_LAST
<MAV_CMD_DO_LAST
in that way we catch all possibility and notify if it is possible to be use or not. "
What do you think?

**Comment by khancyr** [comment]
Maybe I will update only with Randy suggestion an make another PR to catch all cmd

**Comment by khancyr** [comment]
Rebased on master.
All vehicule have the same code.
Invalid message has been simplify as I don't think we need to verbose error more than say that X cmd is invalid.
In plane some case have been move to be in the same order than in start_command function, that way is easier to check if everything is ok

**Comment by magicrub** [comment]
Merged!
